src/loader.js

```diff
@@ -0,0 +1,91 @@
+let assume = require('assume');
+let debug = require('debug')('taskcluster:loader');
+let Promise = require('promise');
+
+function loader(componentDirectory, requiredVirtualComponents) {
+  /**
+   * If present, this is a list of component names that a loader
+   * must define.  These are likely things like configuration values
+   */
+  if (requiredVirtualComponents) {
+    assume(requiredVirtualComponents).is.array();
+  } else {
+    requiredVirtualComponents = [];
+  }
+
```

Comment:
you should also validate that all referenced components is defined... we can do statically...